common/numerated/src/numerated.rs

```diff
@@ -16,22 +16,11 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! [Numerated], [Bound] traits definition and implementation for integer types.
+//! [Numerated], [Bound] traits definition and implementations for integer types.
+//! Also [OptionBound] type is defined, which can be used as [Bound] for any type `T: Numerated`.
 
-use num_traits::{bounds::UpperBounded, One, PrimInt, Unsigned};
-
-/// Represents a value or upper bound. Can be in two states:
-/// - Value: contains value.
-/// - Upper: contains max value for `T`.
-///
-/// See also trait [Bound].
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum BoundValue<T: Sized> {
-    /// The bound is a value. Contains `T` value.
-    Value(T),
-    /// The bound is an upper bound. Contains `T` max value.
-    Upper(T),
-}
+use core::cmp::Ordering;
+use num_traits::{One, PrimInt, Unsigned};
 
 /// For any type `T`, `Bound<T>` is a type, which has set of values bigger than `T` by one element.
 /// - Each value from `T` has unambiguous mapping to `Bound<T>`.
```

Comment:
Should we consider specific enum or struct for `Bound` to force it being a type instead of some strange trait?

Reply:
No, it must be trait to allow user to choose his own bound type. Also you can always use OptionBound, if you do not want to implement yours